apps/fishing-map/features/analysis/AnalysisBeforeAfter.tsx

```diff
@@ -42,12 +44,20 @@ const AnalysisBeforeAfter: React.FC<AnalysisTypeProps> = (props) => {
     <Fragment>
       <AnalysisDescription description={description} />
       <div className={styles.container}>
-        {dataviews && dataviews.map((d) => <DatasetFilterSource dataview={d} hideColor={true} />)}
+        {dataviews &&
+          dataviews.map((d) => <DatasetFilterSource dataview={d} key={d.id} hideColor={true} />)}
       </div>
-      {/*
-        TODO: Draw graph using layersTimeseriesFiltered
-      */}
-      <div className={cx(styles.container, styles.comingSoon)}>Graph coming soon</div>
+      {layersTimeseriesFiltered ? (
+        <AnalysisBeforeAfterGraph
+          graphData={layersTimeseriesFiltered?.[0]}
+          start={timeComparison.start}
+          end={timeComparisonValues.end}
+        />
+      ) : (
+        <div className={styles.graphContainer}>
+          <Spinner />
```

Comment:
this is not always working as expected because first load the `layersTimeseriesFiltered` is undefined but once the dataview is loaded (for example you go to evolution and back to before/after it contains and empty timeseries array
![image](https://user-images.githubusercontent.com/10500650/142887300-9d35d0c0-38e2-4509-b169-6dda58bee4dc.png)

So I guess I'd show the loading when undefined and also when timeseries is empty